File: src/main/java/com/github/eendroroy/sdk/openweather/response/MainResponse.java
package com.github.eendroroy.sdk.openweather.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * @author indrajit
 */

@ToString
@Setter
@Getter
@JsonRootName("main")
public class MainResponse implements BaseResponse {
    @JsonProperty("temp")
    public Float temp;

    @JsonProperty("feels_like")
    public Float feelsLike;

    @JsonProperty("temp_min")
    public Float tempMin;

    @JsonProperty("temp_max")
    public Float tempMax;

    @JsonProperty("pressure")
    public Integer pressure;

    @JsonProperty("humidity")
    public Integer humidity;
}

File: src/main/kotlin/com/github/eendroroy/sdk/openweathermap/response/MainResponse.kt
package com.github.eendroroy.sdk.openweathermap.response

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonRootName

/**
 * @author indrajit
 */
@JsonRootName("main")
class MainResponse : BaseResponse {
    @JsonProperty("temp")
    var temp: Float? = null

    @JsonProperty("feels_like")
    var feelsLike: Float? = null

    @JsonProperty("temp_min")
    var tempMin: Float? = null

    @JsonProperty("temp_max")
    var tempMax: Float? = null

    @JsonProperty("pressure")
    var pressure: Int? = null

    @JsonProperty("humidity")
    var humidity: Int? = null
}